features/step_definitions/logging.rb

```diff
@@ -705,3 +705,37 @@
     end
   end
 end
+
+# Delete the fluentd buffer files in nodes. the fluentd must be deleted before running this step.
+Given /^logging collector bufferfiles are cleared on all nodes$/ do
+  ensure_admin_tagged
+  serviceaccount="collector-clean"
+  ds_name="collector-clean"
+  user(0)
+  step %Q/I have a project/
+
+  fluentd_pods = BushSlicer::Pod.get_labeled("logging-infra=fluentd", project: project("openshift-logging", switch: false), user: admin)
+  if(fluentd_pods.length()>0)
+    puts("The fluetnd pods is still running")
+    @result[":success"] = false
+  else
+    sa = service_account("#{serviceaccount}")
+    @result =  sa.create(by: admin)
+    step %Q/the step should succeed/
+
+    _opts = {scc: "privileged", user_name: "system:serviceaccount:#{project.name}:#{serviceaccount}"}
+    @result = admin.cli_exec(:oadm_policy_add_scc_to_user, **_opts)
+    step %Q/the step should succeed/
+    @result = user.cli_exec(:create, {f: "#{BushSlicer::HOME}/testdata/logging/collecter/collector_clear.json"})
+    step %Q/the step should succeed/
+    
+    pod_num=daemon_set(ds_name).replica_counters[:desired]
+    step %Q/#{pod_num} pods become ready with labels:/, table(%{
+      | logging-infra=#{ds_name} |
+    })
+
+    admin.cli_exec(:delete, object_type: 'daemonset', object_name_or_id: ds_name, n: project.name)
```

Comment:
These 2 objects couldn't be removed if any one of the above steps is failed.
I think we'd better use `teardown_add`:
```
  teardown_add {
    admin.cli_exec(:delete, object_type: 'daemonset', object_name_or_id: ds_name, n: project.name)
    admin.cli_exec(:delete, object_type: 'sa', object_name_or_id: serviceaccount, n: project.name)
  }
```
And I think it's better to put this block above line 729, so these objects could be removed when the scenario finished.

Reply:
The whole project will be deleted after the scenarios. I think we needn't put it in teardown block. If it is in tear down block. the delete steps will not be executed in tail

Reply:
No, the steps in teardown block will be executed after the scenario.

Reply:
but better to use 
```
admin ensures "xxxx" daemon_set is deleted from the "xxxx" project after scenario
```